File: 2026_GFG_POTD/JANUARY/14-01-2026/Police and Thieves.cpp
class Solution {
public:
    int catchThieves(vector<char> &arr, int k) {
        vector<int> police, thieves;
        int n = arr.size(), caught = 0;

        for (int i = 0; i < n; i++) {
            if (arr[i] == 'P') police.push_back(i);
            else if (arr[i] == 'T') thieves.push_back(i);
        }

        int i = 0, j = 0;
        while (i < police.size() && j < thieves.size()) {
            if (abs(police[i] - thieves[j]) <= k) {
                caught++;
                i++; j++;
            } else if (police[i] < thieves[j]) i++;
            else j++;
        }

        return caught;
    }
};

File: 2026_GFG_POTD/JANUARY/14-01-2026/Police and Thieves.cs
using System;
using System.Collections.Generic;

class Solution {
    public int catchThieves(char[] arr, int k) {
        List<int> police = new List<int>();
        List<int> thieves = new List<int>();
        int n = arr.Length;
        int caught = 0;

        // Collect indices of police and thieves
        for (int idx = 0; idx < n; idx++) {
            if (arr[idx] == 'P') police.Add(idx);
            else if (arr[idx] == 'T') thieves.Add(idx);
        }

        int p = 0, t = 0;
        while (p < police.Count && t < thieves.Count) {
            if (Math.Abs(police[p] - thieves[t]) <= k) {
                caught++;
                p++;
                t++;
            } else if (police[p] < thieves[t]) {
                p++;
            } else {
                t++;
            }
        }

        return caught;
    }
}